### ISSUE TITLE
Fix type inferencing and syntax highlighting for lambdas with variable expression for parameter list

### ISSUE THREAD

**Issue by eric-milles** [groovy/groovy-eclipse#1000]
Consider the following:
```groovy
List<String> list = ['a', 'e', 'i', 'o', 'u', 'y']
list = ['abc', 'def', 'ghi', 'xyz'].stream().filter(
  e -> list.stream().anyMatch(c -> e.contains(c))).toList()
```

The variable expressions "e" and "c" are missing name offsets, causing a breakdown in syntax highlighting.  Also the type of each should be inferred as String, not Object.

![image](https://user-images.githubusercontent.com/18193802/70647296-415daf00-1c0e-11ea-8598-3dee832b1833.png)

**Comment by eric-milles** [comment]
![image](https://user-images.githubusercontent.com/18193802/70648664-30626d00-1c11-11ea-8d88-950186e04cf0.png)

**Comment by eric-milles** [comment]
![image](https://user-images.githubusercontent.com/18193802/70734709-5ac82f00-1cd2-11ea-8d0d-02ce6f415316.png)

**Comment by eric-milles** [comment]
![including fix for #1002](https://user-images.githubusercontent.com/18193802/71564924-11b40100-2a6e-11ea-9830-85a4385dcaa4.png)
